src/main/java/teammates/logic/core/CommentsLogic.java

```diff
@@ -101,6 +102,40 @@ public CommentAttributes getComment(Long commentId) {
         return commentsDb.getCommentsForReceiver(courseId, recipientType, receiverEmail);
     }
     
+    /**
+     * Gets comments for a particular receiver, then filters out comments that the instructor cannot see.
+     * @param courseId
+     * @param recipientType
+     * @param receiverEmail
+     * @param instructorEmail
+     * @return List of comments visible to the instructor, directed at the receiver.
+     * @throws EntityDoesNotExistException
+     */
+    public List<CommentAttributes> getCommentsForReceiverVisibleToInstructor(
+            String courseId, CommentParticipantType recipientType, String receiverEmail, String instructorEmail)
+            throws EntityDoesNotExistException {
+        verifyIsCoursePresent(courseId, "get");
+        verifyIsInstructorOfCourse(courseId, instructorEmail);
+        List<CommentAttributes> commentsFromLogic = commentsDb.getCommentsForReceiver(courseId, recipientType, receiverEmail);
+        Iterator<CommentAttributes> iterator = commentsFromLogic.iterator();
+        List<CommentAttributes> comments = new LinkedList<CommentAttributes>();
+        HashSet<String> commentsVisitedSet = new HashSet<String>();
```

Comment:
`preventAppendingThisCommentAgain` requires use of a `HashSet<String>`, and is used by a few other methods. I'll see if it's possible to change those without breaking anything.